_dev/src/components/smart-shopping-campaign-creation/smart-shopping-campaign-creation.vue

```diff
@@ -537,47 +584,86 @@ export default {
-      }]);
+      const filtersForAPI = [
+        {
+          dimension: dimension.id,
+          values: returnChildrenIds(dimension),
+        },
+      ];
+      this.$store.commit('smartShoppingCampaigns/SET_FILTERS_CHOSEN', filtersForAPI);
+      if (this.editMode) {
+        this.foundSsc.productFilters = filtersForAPI;
+      }
     },
     getDatasFiltersDimensions(search) {
       this.loader = true;
-      this.$store.dispatch('smartShoppingCampaigns/GET_DIMENSIONS_FILTERS', search).then(() => {
-        this.loader = false;
-      });
+      this.$store
+        .dispatch('smartShoppingCampaigns/GET_DIMENSIONS_FILTERS', search)
+        .then(() => {
+          this.loader = false;
+        });
+    },
+    setInterfaceForEdition() {
+      let {endDate} = this.foundSsc;
+
+      const todayYear = new Date().getFullYear();
+      if (new Date(endDate).getFullYear() - 10 > todayYear) {
+        endDate = null;
+      }
+      this.campaignName = this.foundSsc.campaignName;
+      this.campaignDurationStartDate = this.foundSsc.startDate;
+      this.campaignDurationEndDate = endDate;
+      this.campaignHasNoProductsFilter = !this.foundSsc.productFilters.length
+          && !this.foundSsc.hasUnhandledFilters;
+      this.campaignDailyBudget = this.foundSsc.dailyBudget;
+      this.campaignIsActive = this.foundSsc.status === CampaignStatus.ELIGIBLE;
+      this.targetCountry = this.$options.filters.changeCountriesCodesToNames([
+        this.foundSsc.targetCountry,
+      ]);
+      this.hasUnhandledFilters = this.foundSsc.hasUnhandledFilters;
+      this.debounceName();
+      this.$store.commit('smartShoppingCampaigns/SET_FILTERS_CHOSEN', this.filtersChosen);
+      if (this.filtersChosen.length) {
+        let dimensionToEdit = this.$store.state.smartShoppingCampaigns.sscAvailableFilters.find(
+          (dim) => dim.id === this.filtersChosen[0].dimension,
+        );
+        const filtersToFind = this.filtersChosen[0].values;
+        dimensionToEdit = findAndCheckFilter(dimensionToEdit, filtersToFind);
+        this.$store.commit('smartShoppingCampaigns/SET_DIMENSION_CHOSEN', dimensionToEdit);
+      }
+
+      this.loader = false;
     },
   },
   watch: {
     campaignName(oldVal, newVal) {
-      if ((newVal !== oldVal) && this.errorCampaignNameExistsAlready !== null) {
-        this.$store.commit('smartShoppingCampaigns/SET_ERROR_CAMPAIGN_NAME_EXISTS', null);
+      if (newVal !== oldVal && this.errorCampaignNameExistsAlready !== null) {
+        this.$store.commit(
+          'smartShoppingCampaigns/SET_ERROR_CAMPAIGN_NAME_EXISTS',
+          null,
+        );
       }
     },
+    $route: {
+      handler(route) {
+        if (route.name === 'campaign-creation') {
+          this.$store.commit('smartShoppingCampaigns/SET_DIMENSION_CHOSEN', {});
+          this.$store.state.smartShoppingCampaigns.sscAvailableFilters.forEach((dim) => {
+            deepCheckDimension(dim, false);
+          });
```

Comment:
What do you think in having this in the mounted with `setInterfaceForCreation`, so we have the same logic?